proto/message_contents/public_key.proto

```diff
@@ -0,0 +1,41 @@
+// Public Key
+syntax = "proto3";
+
+package xmtp.message_contents;
+
+option go_package = "github.com/xmtp/proto/go/message_contents";
+
+// Signature represents a generalized public key signature,
+// defined as a union to support cryptographic algorithm agility.
+message Signature {
```

Comment:
In the future this signature may be used outside of the publicKey message, to sign other objects. I don't think its really worth splitting it out, to figured I would point it out

Reply:
Hm, I think this might be worth splitting out as well, especially as the number of signature and key types grows. We should already have a second Signature type for wallet signatures. So unless there are objections, I'll split it out too.